src/views/transaction-detail/index.js

```diff
@@ -10,21 +10,22 @@ import TBody from '../../components/public/table/tbody';
 import none from '../../utils/images/none.png';
 import success from '../../utils/images/check_green.png';
 import failed from '../../utils/images/wrong_red.png';
-import Config from "../../config/config.json"
+import {Config} from "../../config/config"
 import {
     withRouter
 } from 'react-router-dom'
 import PocketService from "../../core/services/pocket-service";
-import {DataSource} from "../../datasource/datasource";
+import {getDataSource} from "../../datasource";
 
-const dataSource = new DataSource();
+const dataSource = getDataSource();
 
 class TransactionDetail extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
             txHash: this.props.location.data,
+            loadFromCache: this.props.location.loadFromCache,
```

Comment:
Oof, this could probably be null or give an error, depending on how the URL is copied. Can't we destructure and assign this in `componentWillMount`? Feels like something _can_ explode here.

Reply:
If this is all old code, we can ignore for now